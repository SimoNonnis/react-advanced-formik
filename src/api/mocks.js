export const mockVendorsList = [
  {
    id: 4,
    primary: true,
    type: 'legal_owner',
    present_at_home_visit: true,
    title: 'Dr.',
    first_names: 'Tim',
    last_name: 'Lewis',
    email_address: 'noah83@example.com',
    phone_number: '07000000000',
    date_of_birth: '1975-01-01',
    occupation: 'developer',
    politically_exposed: null,
    living_at_the_property_you_are_selling: false,
    living_at_the_property_you_are_selling_for_at_least_twelve_months: null,
    country_of_residence: 'United Kingdom',
    addresses: [
      {
        id: 1,
        sub_building: null,
        building_name: null,
        building_number: '68',
        street: 'Bell Brook',
        town: 'West Damien',
        county: 'Suffolk',
        postcode: 'SE26 6JG',
        position: 0,
      },
      {
        id: 2,
        sub_building: null,
        building_name: null,
        building_number: '617',
        street: 'Frank Plaza',
        town: 'Carolefurt',
        county: 'Suffolk',
        postcode: 'AB11 5QE',
        position: 1,
      },
    ],
    owned_since_month: null,
    owned_since_year: null,
    reason_for_selling: 'Selling to buy a larger house',
  },
  {
    id: 14,
    primary: false,
    type: 'beneficial_owner',
    present_at_home_visit: true,
    title: 'Mr.',
    first_names: 'New',
    last_name: 'Person',
    email_address: null,
    phone_number: null,
    date_of_birth: null,
    occupation: null,
    politically_exposed: null,
    living_at_the_property_you_are_selling: null,
    living_at_the_property_you_are_selling_for_at_least_twelve_months: null,
    country_of_residence: null,
    addresses: [],
    owned_since_month: null,
    owned_since_year: null,
    reason_for_selling: null,
  },
];

export const mockVendor = {
  id: 4,
  primary: true,
  type: 'legal_owner',
  present_at_home_visit: true,
  title: 'Mr.',
  first_names: 'Simon',
  last_name: 'Nonnis',
};

export const mockAddress = {
  sub_building: 'Unit 1A',
  building_name: 'The Building',
  building_number: '14',
  street: 'Church Field Road',
  town: 'Sudbury',
  county: 'Suffolk',
  postcode: 'CO10 2YA',
  position: 0,
};

export const mockAddressesList = [
  { Id: '52509479.00', StreetAddress: 'P C A Predict', Place: 'Waterside Basin Road Worcester' },
  { Id: '53327590.00', StreetAddress: 'Lock Keepers Cottage Basin Road', Place: 'Worcester' },
  { Id: '26772356.00', StreetAddress: 'Lock View Basin Road', Place: 'Worcester' },
];

export const mockAddressInfo = [
  {
    Udprn: '52509479',
    Company: 'P C A Predict',
    Department: '',
    Line1: 'Waterside',
    Line2: 'Basin Road',
    Line3: '',
    Line4: '',
    Line5: '',
    PostTown: 'Worcester',
    County: 'Worcestershire',
    Postcode: 'WR5 3DA',
    Mailsort: '94142',
    Barcode: '(WR53DA1PX)',
    Type: 'SmallBusiness',
    DeliveryPointSuffix: '1P',
    SubBuilding: '',
    BuildingName: 'Waterside',
    BuildingNumber: '',
    PrimaryStreet: 'Basin Road',
    SecondaryStreet: '',
    DoubleDependentLocality: '',
    DependentLocality: '',
    PoBox: '',
    PrimaryStreetName: 'Basin',
    PrimaryStreetType: 'Road',
    SecondaryStreetName: '',
    SecondaryStreetType: '',
    CountryName: 'England',
    Location: { id: null, latitude: 52.1829787, longitude: -2.2216089 },
  },
];

export const mockDocumentTypeList = {
  identification_list_a: [
    {
      list: 'A',
      type: 'proof_of_id_uk_passport',
      label: 'UK Passport',
      description: 'Current valid (signed) full UK Passport',
    },
  ],
  identification_list_b: [
    {
      list: 'B',
      type: 'proof_of_id_bank_society_or_union_statement',
      label: 'Bank Statement',
      description:
        'Bank, Building Society or Credit Union Statement dated within the last 3 months (not printed off the internet)',
    },
  ],
  proof_of_address_list_c: [
    {
      list: 'C',
      type: 'proof_of_address_bank_statement',
      label: 'Bank Statement',
      description: 'Bank Statement (dated in the last three months) â€“ may be an e-copy',
    },
  ],
};
