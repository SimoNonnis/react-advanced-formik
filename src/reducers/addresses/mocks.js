import { Map, List } from 'immutable';

export const mockedState = Map({
  addresses: Map({
    '5': List([]),
    '35': List([Map({ id: 1 })]),
    '4': List([Map({ id: 1 }), Map({ id: 2 })]),
  }),
  addressesList: [],
  isDisabled: false,
  isLoading: false,
});

export const mockedAddressesList = [
  {
    Id: 100,
    StreetAddress: 'Test street',
    Place: 'anywhere',
  },
];

export const mockedAddressInfo = [
  {
    SubBuilding: 'Test',
    BuildingName: 'Test',
    BuildingNumber: 'Test',
    PrimaryStreet: 'Test',
    PostTown: 'Test',
    County: 'Test',
    Postcode: 'Test',
    Test: 'Test',
  },
  {},
  {},
];

export const mockedAddress = {
  id: 2,
  sub_building: null,
  building_name: null,
  building_number: '892',
  street: 'Alexander Junctions',
  town: 'Graceview',
  county: 'Suffolk',
  postcode: 'WF9 2JY',
  position: 1,
};
