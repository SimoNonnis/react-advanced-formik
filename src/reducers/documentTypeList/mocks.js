export const mockedDocumentTypeList = {
  identification_list_a: [
    {
      list: 'A',
      type: 'proof_of_id_uk_passport',
      label: 'UK Passport',
      description: 'Current valid (signed) full UK Passport',
    },
    {
      list: 'A',
      type: 'proof_of_id_overseas_passport',
      label: 'Overseas Passport',
      description: 'Current valid (signed) overseas Passport',
    },
    {
      list: 'A',
      type: 'proof_of_id_uk_driving_licence',
      label: 'UK Photocard Driving Licence',
      description: 'Current UK photocard Driving Licence (provisional acceptable)',
    },
    {
      list: 'A',
      type: 'proof_of_id_eu_driving_licence',
      label: 'EU Photocard Driving Licence',
      description: 'Current EU photocard Driving Licence',
    },
    {
      list: 'A',
      type: 'proof_of_id_eea_state_id',
      label: 'EEA Member State ID Card',
      description: 'Current valid EEA Member State ID card',
    },
    {
      list: 'A',
      type: 'proof_of_id_biometric_residence_permit',
      label: 'Biometric Residence Permit',
      description: 'Current biometric residence permit issued by UK Border Agency',
    },
    {
      list: 'A',
      type: 'proof_of_id_firearm_shotgun_certificate',
      label: 'Current Firearms Certificate',
      description: 'Current Firearms/Shotgun Certificate',
    },
    {
      list: 'A',
      type: 'proof_of_id_uk_military_card',
      label: 'UK Military Identity Card',
      description: 'Current UK Military Identity Card',
    },
  ],
  identification_list_b: [
    {
      list: 'B',
      type: 'proof_of_id_bank_society_or_union_statement',
      label: 'Bank Statement',
      description:
        'Bank, Building Society or Credit Union Statement dated within the last 3 months (not printed off the internet)',
    },
    {
      list: 'B',
      type: 'proof_of_id_benefits_or_pensions_notification_letter',
      label: 'Benefits or Pensions Letter',
      description: 'Benefits or pensions notification letter confirming the right to benefit',
    },
    {
      list: 'B',
      type: 'proof_of_id_blue_disabled_drivers_pass',
      label: 'Blue Disabled Drivers Pass',
      description: 'Blue disabled drivers pass',
    },
    {
      list: 'B',
      type: 'proof_of_id_current_uk_paper_driving_licence',
      label: 'UK Paper Driving Licence',
      description: 'Current UK paper driving licence',
    },
    {
      list: 'B',
      type: 'proof_of_id_local_authority_tax_bill_or_council_tax_bill',
      label: 'Tax Bill',
      description:
        'Local authority tax bill/council tax bill (we can only accept bills dated until the end of June of the year the client contract is signed)',
    },
    {
      list: 'B',
      type: 'proof_of_id_medical_card_or_certificate',
      label: 'Medical Card/Certificate',
      description: 'Medical Card/Certificate',
    },
    {
      list: 'B',
      type: 'proof_of_id_national_insurance_card',
      label: 'National Insurance Card',
      description: 'National Insurance Card',
    },
    {
      list: 'B',
      type: 'proof_of_id_uk_birth_certificate',
      label: 'UK Birth Certificate',
      description: 'UK Birth Certificate',
    },
    {
      list: 'B',
      type: 'proof_of_id_utility_bill_utility_statement_or_certificate_letter',
      label: 'Utility Bill',
      description:
        'Utility Bill/Utility Statement or Certificate/Letter from a supplier of utilities dated within the last 3 months',
    },
  ],
  proof_of_address_list_c: [
    {
      list: 'C',
      type: 'proof_of_address_bank_statement',
      label: 'Bank Statement',
      description: 'Bank Statement (dated in the last three months) – may be an e-copy',
    },
    {
      list: 'C',
      type: 'proof_of_address_credit_card_statement',
      label: 'Credit Card Statement',
      description: 'Credit Card Statement (dated in the last three months) – may be an e-copy',
    },
    {
      list: 'C',
      type: 'proof_of_address_council_tax_bill',
      label: 'Council Tax Bill',
      description:
        'Council Tax bill (we can only accept bills dated until the end of June of the year the client contract is signed)',
    },
    {
      list: 'C',
      type: 'proof_of_address_current_mortgage_statement',
      label: 'Mortgage Statement',
      description:
        'Current mortgage statement (correspondence address and address the mortgage applies to must be the same)',
    },
    {
      list: 'C',
      type: 'proof_of_address_current_tv_licence',
      label: 'TV Licence',
      description: 'Current TV licence',
    },
    {
      list: 'C',
      type: 'proof_of_address_driving_licence_with_current_address',
      label: 'Driving Licence',
      description:
        'Driving licence showing current address (paper OR card version) (provisional acceptable) (not acceptable if used as ID)',
    },
    {
      list: 'C',
      type: 'proof_of_address_current_home_insurance_policy',
      label: 'Home Insurance Policy',
      description: 'Homeowner’s current home insurance policy schedule',
    },
    {
      list: 'C',
      type: 'proof_of_address_home_service_provider_bill',
      label: 'Broadband or Digital TV Bill',
      description:
        'Home service provider bill, such as broadband or digital TV dated within the last three months',
    },
    {
      list: 'C',
      type: 'proof_of_address_letter_from_the_employer_on_company',
      label: 'Letter from the Employer',
      description:
        'Letter from the employer on company headed paper, signed and dated within the last three months (an email from a verified company email address is acceptable)',
    },
    {
      list: 'C',
      type: 'proof_of_address_house_benefits_documentation',
      label: 'Housing Allowance/Benefit',
      description:
        'Recent documentation confirming the applicant will be/is receiving local housing allowance or housing benefit',
    },
    {
      list: 'C',
      type: 'proof_of_address_tenancy_agreement',
      label: 'Tenancy Agreement',
      description:
        'Tenancy agreement signed and dated within the last six months (the customer being checked must be a named tenant on the tenancy agreement)',
    },
    {
      list: 'C',
      type: 'proof_of_address_utility_bill',
      label: 'Utility Bill',
      description: 'Utility bill dated within the last three months',
    },
    {
      list: 'C',
      type: 'proof_of_address_letter_from_the_nhs',
      label: 'Letter from the NHS',
      description: 'Letter from the NHS writing to confirm the customer is living at the address',
    },
    {
      list: 'C',
      type: 'proof_of_address_letter_from_bank',
      label: 'Letter from the Bank',
      description:
        'Letter from bank to confirm the customer is living at the property – no other bank letter is acceptable',
    },
  ],
};
