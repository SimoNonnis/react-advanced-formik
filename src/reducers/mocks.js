export const mockedVendors = [
  {
    id: 4,
    primary: true,
    type: 'legal_owner',
    present_at_home_visit: true,
    title: 'Mr',
    first_names: 'Finley',
    last_name: 'Powell',
    email_address: 'lucas.graham@example.net',
    phone_number: '07000000000',
    date_of_birth: '1974-12-01',
    occupation: 'developer',
    politically_exposed: true,
    living_at_the_property_you_are_selling: false,
    living_at_the_property_you_are_selling_for_at_least_twelve_months: null,
    country_of_residence: 'United Kingdom',
    addresses: [
      {
        id: 1,
        sub_building: null,
        building_name: null,
        building_number: '202',
        street: 'Amelia Courts',
        town: 'Lake Staceyview',
        county: 'Suffolk',
        postcode: 'DD1 2NF',
        position: 0,
      },
      {
        id: 2,
        sub_building: null,
        building_name: null,
        building_number: '86',
        street: 'Evelyn Springs',
        town: 'Walkermouth',
        county: 'Suffolk',
        postcode: 'SG6 1PR',
        position: 1,
      },
    ],
    owned_since_month: 'January',
    owned_since_year: '2000',
    reason_for_selling: 'Selling to buy a larger house',
    proof_of_address_files: [
      {
        id: 2,
        list: 'C',
        type: 'proof_of_address_bank_statement',
        label: 'Bank Statement',
        mime_type: 'image/jpeg',
        path: '/admin/check/5/fetch-file/2',
      },
    ],
    identification_files: [
      {
        id: 8,
        list: 'B',
        type: 'proof_of_id_bank_society_or_union_statement',
        label: 'Bank Statement',
        mime_type: 'image/png',
        path: '/admin/check/7/fetch-file/8',
      },
      {
        id: 9,
        list: 'B',
        type: 'proof_of_id_blue_disabled_drivers_pass',
        label: 'Blue Disabled Drivers Pass',
        mime_type: 'image/jpeg',
        path: '/admin/check/7/fetch-file/9',
      },
    ],
  },
  {
    id: 34,
    primary: false,
    type: 'legal_owner',
    present_at_home_visit: false,
    title: 'Mr',
    first_names: 'Ray',
    last_name: 'Ross',
    email_address: 'danielle92@example.net',
    phone_number: '07000000000',
    date_of_birth: '1996-01-24',
    occupation: 'developer',
    politically_exposed: false,
    living_at_the_property_you_are_selling: false,
    living_at_the_property_you_are_selling_for_at_least_twelve_months: null,
    country_of_residence: 'France',
    addresses: [
      {
        id: 5,
        sub_building: null,
        building_name: null,
        building_number: '92',
        street: 'Dave Drive',
        town: 'Bellview',
        county: 'Suffolk',
        postcode: 'G65 9BG',
        position: 0,
      },
    ],
    owned_since_month: 'January',
    owned_since_year: '2000',
    reason_for_selling: 'Selling to buy a larger house',
    proof_of_address_files: [
      {
        id: 5,
        list: 'C',
        type: 'proof_of_address_current_tv_licence',
        label: 'TV Licence',
        mime_type: 'image/jpeg',
        path: '/admin/check/6/fetch-file/5',
      },
    ],
    identification_files: [
      {
        id: 3,
        list: 'B',
        type: 'proof_of_id_uk_birth_certificate',
        label: 'UK Birth Certificate',
        mime_type: 'image/jpeg',
        path: '/admin/check/8/fetch-file/3',
      },
      {
        id: 4,
        list: 'B',
        type: 'proof_of_id_utility_bill_utility_statement_or_certificate_letter',
        label: 'Utility Bill',
        mime_type: 'image/jpeg',
        path: '/admin/check/8/fetch-file/4',
      },
    ],
  },
];

export const mockedVendor = {
  id: 55,
  primary: false,
  type: 'legal_owner',
  present_at_home_visit: true,
  title: 'mr',
  first_names: 'Jack',
  last_name: 'London',
  email_address: null,
  phone_number: null,
  date_of_birth: null,
  occupation: null,
  politically_exposed: null,
  living_at_the_property_you_are_selling: true,
  living_at_the_property_you_are_selling_for_at_least_twelve_months: true,
  country_of_residence: 'United Kingdom',
  addresses: [
    {
      id: 143,
      sub_building: null,
      building_name: null,
      building_number: '4',
      street: 'Victoria Tunnel',
      town: 'Port Louisberg',
      county: 'Suffolk',
      postcode: 'RG14 2EL',
      position: 0,
    },
  ],
  owned_since_month: null,
  owned_since_year: null,
  reason_for_selling: null,
  proof_of_address_files: [
    {
      id: 11,
      list: 'C',
      type: 'proof_of_address_current_mortgage_statement',
      label: 'Mortgage Statement',
      mime_type: 'image/png',
      path: '/admin/check/10/fetch-file/11',
    },
  ],
  identification_files: [
    {
      id: 6,
      list: 'A',
      type: 'proof_of_id_overseas_passport',
      label: 'Overseas Passport',
      mime_type: 'image/png',
      path: '/admin/check/9/fetch-file/6',
    },
  ],
};
