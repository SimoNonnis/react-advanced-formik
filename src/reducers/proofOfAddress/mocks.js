import { Map, List } from 'immutable';

export const mockedState = Map({
  vendorsProofOfAddress: Map({
    '4': List([
      Map({
        id: 2,
        list: 'C',
        type: 'proof_of_address_bank_statement',
        label: 'Bank Statement',
        mime_type: 'image/jpeg',
        path: '/admin/check/5/fetch-file/2',
      }),
    ]),
    '34': List([]),
  }),
  isLoading: Map({
    '4': false,
    '34': false,
  }),
});

export const mockedPoAd = {
  id: 40,
  list: 'C',
  type: 'proof_of_address_current_mortgage_statement',
  label: 'Mortgage Statement',
  mime_type: 'image/jpeg',
  path: '/admin/check/5/fetch-file/2',
};
