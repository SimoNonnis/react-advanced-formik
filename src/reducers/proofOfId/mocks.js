import { Map, List } from 'immutable';

export const mockedState = Map({
  vendorsProofOfId: Map({
    '4': List([
      Map({
        id: 8,
        list: 'B',
        type: 'proof_of_id_bank_society_or_union_statement',
        label: 'Bank Statement',
        mime_type: 'image/png',
        path: '/admin/check/7/fetch-file/8',
      }),
    ]),
    '34': List([]),
  }),
  isLoading: Map({
    '4': false,
    '34': false,
  }),
});

export const mockedPoId = {
  id: 80,
  list: 'A',
  type: 'proof_of_id_overseas_passport',
  label: 'Overseas Passport',
  mime_type: 'image/png',
  path: '/admin/check/7/fetch-file/8',
};
