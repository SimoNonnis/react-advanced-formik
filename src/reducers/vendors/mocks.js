import { Map, List } from 'immutable';
import { mockedVendors, mockedVendor } from 'reducers/mocks';
import { initialState } from './';

export const mockedRequestState = initialState.merge(Map({ isFetchingVendors: true }));

export const mockedState = Map({
  isFetchingVendors: false,
  vendors: List([...mockedVendors]),
});

export const mockedStateAfterDelete = Map({
  isFetchingVendors: false,
  vendors: List([
    {
      id: 4,
      primary: true,
      type: 'legal_owner',
      present_at_home_visit: true,
      title: 'Mr',
      first_names: 'Finley',
      last_name: 'Powell',
      email_address: 'lucas.graham@example.net',
      phone_number: '07000000000',
      date_of_birth: '1974-12-01',
      occupation: 'developer',
      politically_exposed: true,
      living_at_the_property_you_are_selling: false,
      living_at_the_property_you_are_selling_for_at_least_twelve_months: null,
      country_of_residence: 'United Kingdom',
      addresses: [
        {
          id: 1,
          sub_building: null,
          building_name: null,
          building_number: '202',
          street: 'Amelia Courts',
          town: 'Lake Staceyview',
          county: 'Suffolk',
          postcode: 'DD1 2NF',
          position: 0,
        },
        {
          id: 2,
          sub_building: null,
          building_name: null,
          building_number: '86',
          street: 'Evelyn Springs',
          town: 'Walkermouth',
          county: 'Suffolk',
          postcode: 'SG6 1PR',
          position: 1,
        },
      ],
      owned_since_month: 'January',
      owned_since_year: '2000',
      reason_for_selling: 'Selling to buy a larger house',
      proof_of_address_files: [
        {
          id: 2,
          list: 'C',
          type: 'proof_of_address_bank_statement',
          label: 'Bank Statement',
          mime_type: 'image/jpeg',
          path: '/admin/check/5/fetch-file/2',
        },
      ],
      identification_files: [
        {
          id: 8,
          list: 'B',
          type: 'proof_of_id_bank_society_or_union_statement',
          label: 'Bank Statement',
          mime_type: 'image/png',
          path: '/admin/check/7/fetch-file/8',
        },
        {
          id: 9,
          list: 'B',
          type: 'proof_of_id_blue_disabled_drivers_pass',
          label: 'Blue Disabled Drivers Pass',
          mime_type: 'image/jpeg',
          path: '/admin/check/7/fetch-file/9',
        },
      ],
    },
  ]),
});

export const mockedStateAfterAdd = Map({
  isFetchingVendors: false,
  vendors: List([...mockedVendors, mockedVendor]),
});
