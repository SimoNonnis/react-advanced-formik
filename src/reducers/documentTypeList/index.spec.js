import {
  requestGetDocumentTypeList,
  successGetDocumentTypeList,
  rejectGetDocumentTypeList,
} from 'actions/documents';

import { mockedDocumentTypeList } from './mocks';
import documentTypeListReducer, { initialState } from './';

describe('Test documentTypeListReducer', () => {
  it('should return the initial state', () => {
    expect(documentTypeListReducer(undefined, {})).toBe(initialState);
  });
  it('should set isFetchingDocumentTypeList to true when calling requestGetDocumentTypeList()', () => {
    const reducer = documentTypeListReducer(initialState, requestGetDocumentTypeList());
    expect(reducer.get('isFetchingDocumentTypeList')).toBe(true);
  });
  it('should set isFetchingDocumentTypeList to false when calling successGetDocumentTypeList()', () => {
    const reducer = documentTypeListReducer(initialState, successGetDocumentTypeList());
    expect(reducer.get('isFetchingDocumentTypeList')).toBe(false);
  });
  it('should populate documentTypeList with payload when calling successGetDocumentTypeList()', () => {
    const reducer = documentTypeListReducer(
      initialState,
      successGetDocumentTypeList(mockedDocumentTypeList)
    );
    expect(reducer).toMatchSnapshot();
  });
  it('should set isFetchingDocumentTypeList to false when calling rejectGetDocumentTypeList()', () => {
    const reducer = documentTypeListReducer(initialState, rejectGetDocumentTypeList());
    expect(reducer.get('isFetchingDocumentTypeList')).toBe(false);
  });
});
