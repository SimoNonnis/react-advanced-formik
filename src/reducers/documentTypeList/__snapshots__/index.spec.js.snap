// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test documentTypeListReducer should populate documentTypeList with payload when calling successGetDocumentTypeList() 1`] = `
Object {
  "documentTypeList": Object {
    "identification_list_a": Array [
      Object {
        "description": "Current valid (signed) full UK Passport",
        "label": "UK Passport",
        "list": "A",
        "type": "proof_of_id_uk_passport",
      },
      Object {
        "description": "Current valid (signed) overseas Passport",
        "label": "Overseas Passport",
        "list": "A",
        "type": "proof_of_id_overseas_passport",
      },
      Object {
        "description": "Current UK photocard Driving Licence (provisional acceptable)",
        "label": "UK Photocard Driving Licence",
        "list": "A",
        "type": "proof_of_id_uk_driving_licence",
      },
      Object {
        "description": "Current EU photocard Driving Licence",
        "label": "EU Photocard Driving Licence",
        "list": "A",
        "type": "proof_of_id_eu_driving_licence",
      },
      Object {
        "description": "Current valid EEA Member State ID card",
        "label": "EEA Member State ID Card",
        "list": "A",
        "type": "proof_of_id_eea_state_id",
      },
      Object {
        "description": "Current biometric residence permit issued by UK Border Agency",
        "label": "Biometric Residence Permit",
        "list": "A",
        "type": "proof_of_id_biometric_residence_permit",
      },
      Object {
        "description": "Current Firearms/Shotgun Certificate",
        "label": "Current Firearms Certificate",
        "list": "A",
        "type": "proof_of_id_firearm_shotgun_certificate",
      },
      Object {
        "description": "Current UK Military Identity Card",
        "label": "UK Military Identity Card",
        "list": "A",
        "type": "proof_of_id_uk_military_card",
      },
    ],
    "identification_list_b": Array [
      Object {
        "description": "Bank, Building Society or Credit Union Statement dated within the last 3 months (not printed off the internet)",
        "label": "Bank Statement",
        "list": "B",
        "type": "proof_of_id_bank_society_or_union_statement",
      },
      Object {
        "description": "Benefits or pensions notification letter confirming the right to benefit",
        "label": "Benefits or Pensions Letter",
        "list": "B",
        "type": "proof_of_id_benefits_or_pensions_notification_letter",
      },
      Object {
        "description": "Blue disabled drivers pass",
        "label": "Blue Disabled Drivers Pass",
        "list": "B",
        "type": "proof_of_id_blue_disabled_drivers_pass",
      },
      Object {
        "description": "Current UK paper driving licence",
        "label": "UK Paper Driving Licence",
        "list": "B",
        "type": "proof_of_id_current_uk_paper_driving_licence",
      },
      Object {
        "description": "Local authority tax bill/council tax bill (we can only accept bills dated until the end of June of the year the client contract is signed)",
        "label": "Tax Bill",
        "list": "B",
        "type": "proof_of_id_local_authority_tax_bill_or_council_tax_bill",
      },
      Object {
        "description": "Medical Card/Certificate",
        "label": "Medical Card/Certificate",
        "list": "B",
        "type": "proof_of_id_medical_card_or_certificate",
      },
      Object {
        "description": "National Insurance Card",
        "label": "National Insurance Card",
        "list": "B",
        "type": "proof_of_id_national_insurance_card",
      },
      Object {
        "description": "UK Birth Certificate",
        "label": "UK Birth Certificate",
        "list": "B",
        "type": "proof_of_id_uk_birth_certificate",
      },
      Object {
        "description": "Utility Bill/Utility Statement or Certificate/Letter from a supplier of utilities dated within the last 3 months",
        "label": "Utility Bill",
        "list": "B",
        "type": "proof_of_id_utility_bill_utility_statement_or_certificate_letter",
      },
    ],
    "proof_of_address_list_c": Array [
      Object {
        "description": "Bank Statement (dated in the last three months) – may be an e-copy",
        "label": "Bank Statement",
        "list": "C",
        "type": "proof_of_address_bank_statement",
      },
      Object {
        "description": "Credit Card Statement (dated in the last three months) – may be an e-copy",
        "label": "Credit Card Statement",
        "list": "C",
        "type": "proof_of_address_credit_card_statement",
      },
      Object {
        "description": "Council Tax bill (we can only accept bills dated until the end of June of the year the client contract is signed)",
        "label": "Council Tax Bill",
        "list": "C",
        "type": "proof_of_address_council_tax_bill",
      },
      Object {
        "description": "Current mortgage statement (correspondence address and address the mortgage applies to must be the same)",
        "label": "Mortgage Statement",
        "list": "C",
        "type": "proof_of_address_current_mortgage_statement",
      },
      Object {
        "description": "Current TV licence",
        "label": "TV Licence",
        "list": "C",
        "type": "proof_of_address_current_tv_licence",
      },
      Object {
        "description": "Driving licence showing current address (paper OR card version) (provisional acceptable) (not acceptable if used as ID)",
        "label": "Driving Licence",
        "list": "C",
        "type": "proof_of_address_driving_licence_with_current_address",
      },
      Object {
        "description": "Homeowner’s current home insurance policy schedule",
        "label": "Home Insurance Policy",
        "list": "C",
        "type": "proof_of_address_current_home_insurance_policy",
      },
      Object {
        "description": "Home service provider bill, such as broadband or digital TV dated within the last three months",
        "label": "Broadband or Digital TV Bill",
        "list": "C",
        "type": "proof_of_address_home_service_provider_bill",
      },
      Object {
        "description": "Letter from the employer on company headed paper, signed and dated within the last three months (an email from a verified company email address is acceptable)",
        "label": "Letter from the Employer",
        "list": "C",
        "type": "proof_of_address_letter_from_the_employer_on_company",
      },
      Object {
        "description": "Recent documentation confirming the applicant will be/is receiving local housing allowance or housing benefit",
        "label": "Housing Allowance/Benefit",
        "list": "C",
        "type": "proof_of_address_house_benefits_documentation",
      },
      Object {
        "description": "Tenancy agreement signed and dated within the last six months (the customer being checked must be a named tenant on the tenancy agreement)",
        "label": "Tenancy Agreement",
        "list": "C",
        "type": "proof_of_address_tenancy_agreement",
      },
      Object {
        "description": "Utility bill dated within the last three months",
        "label": "Utility Bill",
        "list": "C",
        "type": "proof_of_address_utility_bill",
      },
      Object {
        "description": "Letter from the NHS writing to confirm the customer is living at the address",
        "label": "Letter from the NHS",
        "list": "C",
        "type": "proof_of_address_letter_from_the_nhs",
      },
      Object {
        "description": "Letter from bank to confirm the customer is living at the property – no other bank letter is acceptable",
        "label": "Letter from the Bank",
        "list": "C",
        "type": "proof_of_address_letter_from_bank",
      },
    ],
  },
  "isFetchingDocumentTypeList": false,
}
`;
